mesonbuild/modules/pkgconfig.py

```diff
@@ -120,8 +120,15 @@ def _process_libs(self, libs, public):
                 processed_reqs.append(obj.generated_pc)
             elif isinstance(obj, dependencies.PkgConfigDependency):
                 if obj.found():
-                    processed_reqs.append(obj.name)
-                    self.add_version_reqs(obj.name, obj.version_reqs)
+                    if obj.static:
+                        # in case of embedded static dependency
+                        for arg in obj.link_args:
+                            if not os.sep in arg:
```

Comment:
Why this check? I guess it's because static libs are replaced by full path to .a file, and shared libraries are kept as -lfoo arg? Please add a comment explaining it. I would rather check with `os.path.isabs()` or `endswith('.a')` because pc file could contain / on Windows I guess?